kubectl/service.go

```diff
@@ -86,11 +86,15 @@ func GetLoadBalancerNameFromService(service corev1.Service) (string, error) {
 
 	// TODO: When expanding to GCP, update this logic
 
-	// For ELB, the subdomain will be NAME-TIME
+	// For ELB, the subdomain will be one of NAME-TIME or internal-NAME-TIME
 	loadbalancerHostnameSubDomain := strings.Split(loadbalancerHostname, ".")[0]
 	loadbalancerHostnameSubDomainParts := strings.Split(loadbalancerHostnameSubDomain, "-")
-	if len(loadbalancerHostnameSubDomainParts) != 2 {
+	numParts := len(loadbalancerHostnameSubDomainParts)
+	if numParts == 2 {
+		return loadbalancerHostnameSubDomainParts[0], nil
+	} else if numParts == 3 {
+		return loadbalancerHostnameSubDomainParts[1], nil
+	} else {
 		return "", NewLoadBalancerNameFormatError(loadbalancerHostname)
 	}
```

Comment:
This looks fine and would work, but perhaps a `switch` would be slightly easier to read?

```
switch len(loadbalancerHostnameSubDomainParts) {
case 2:
		return loadbalancerHostnameSubDomainParts[0], nil
case 3:
		return loadbalancerHostnameSubDomainParts[1], nil
default:
		return "", NewLoadBalancerNameFormatError(loadbalancerHostname)
}
```

Reply:
I think that will come down to personal preference. I personally like using `switch` when the if conditions are more than 2, since I find `if`, `else if`, `else` easier to read than a switch statement. But that is mostly because I come from a `python` background where there is no `switch`.

Normally I would be fine to switch (no pun intended 😄 ) because I don't really care, but in this case I'm going to keep as is since I don't want to go through another test cycle for this change and we have a customer waiting. Testing this is more expensive than typical. I have to (a) wait for build, (b) cut a new release and wait for binary, (c) update eks repo and wait for eks test.

Reply:
Understandable! I mainly just wanted to have something useful to contribute. In the end this makes little to no difference in this case. Thanks for putting together the fix so quickly!